src/DelegatingHydratorFactory.php

```diff
@@ -9,21 +9,35 @@
 
 namespace Zend\Hydrator;
 
+use Interop\Container\ContainerInterface;
 use Zend\ServiceManager\FactoryInterface;
 use Zend\ServiceManager\ServiceLocatorInterface;
 
 class DelegatingHydratorFactory implements FactoryInterface
 {
     /**
-     * Creates DelegatingHydrator
+     * Creates DelegatingHydrator (v2)
      *
      * @param  ServiceLocatorInterface $serviceLocator
      * @return DelegatingHydrator
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
+        return $this($serviceLocator, '');
+    }
+
+    /**
+     * Creates DelegatingHydrator (v3)
+     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param array|null $options
+     * @return DelegatingHydrator
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
         // Assume that this factory is registered with the HydratorManager,
         // and just pass it directly on.
-        return new DelegatingHydrator($serviceLocator);
+        return new DelegatingHydrator($container);
```

Comment:
There's a subtle issue here: with current v2, the container passed to the factory will be the `HydratorPluginManager` instance, while with v3, it will be the parent container. In the latter case, `DelegatingHydrator` usage breaks, as it will be attempting to fetch hydrators from the container, but the container will be the application-level container!

Which poses a quandary: how do we get access to the `HydratorPluginManager`? My suggestion is:

- Check to see if `$container` is a `HydratorPluginManager` instance; if so, done.
- Check to see if `$container->has(HydratorPluginManager::class)`; if so, pull that, and done.
- Check to see if `$container->has('HydratorManager')` (the name used by zend-mvc); if so, pull that, and done.
- Finally, if all the above fail, create a new instance of `HydratorPluginManager` and use that.

I'll make that change on merge.